supervised/utils/additional_metrics.py

```diff
@@ -137,6 +137,79 @@ def binary_classification(target, predictions, sample_weight=None):
 
         threshold = float(max_metrics["accuracy"]["threshold"])
 
+        accu_threshold_details = {
+            "f1": [],
+            "accuracy": [],
+            "precision": [],
+            "recall": [],
+            "mcc": [],
+        }
+
+        for i in range(STEPS):
+            if (
+                np.sum(
+                    predictions > details["threshold"][np.argmax(details["accuracy"])]
+                )
+                < 1
+            ):
+                break
+            response = (
+                predictions > details["threshold"][np.argmax(details["accuracy"])]
+            ).astype(int)
+
+            accu_threshold_details["f1"] += [
+                f1_score(target, response, sample_weight=sample_weight)
+            ]
+            accu_threshold_details["accuracy"] += [
+                accuracy_score(target, response, sample_weight=sample_weight)
+            ]
+            accu_threshold_details["precision"] += [
+                precision_score(target, response, sample_weight=sample_weight)
+            ]
+            accu_threshold_details["recall"] += [
+                recall_score(target, response, sample_weight=sample_weight)
+            ]
+            if i == 0:
+                accu_threshold_details["mcc"] += [0.0]
+            else:
+                accu_threshold_details["mcc"] += [
+                    matthews_corrcoef(target, response, sample_weight=sample_weight)
+                ]
+
+        # accuracy_threshold_max_metric metrics
+        accuracy_threshold_max_metric = {
+            "logloss": {
+                "score": logloss(target, predictions, sample_weight=sample_weight),
+                "threshold": None,
+            },  # there is no threshold for LogLoss
+            "auc": {
+                "score": roc_auc_score(
+                    target, predictions, sample_weight=sample_weight
+                ),
+                "threshold": None,
+            },  # there is no threshold for AUC
+            "f1": {
+                "score": np.max(accu_threshold_details["f1"]),
+                "threshold": details["threshold"][np.argmax(details["accuracy"])],
```

Comment:
maybe it is worth to add variable `accuracy_threshold` instead of re-running `details["threshold"][np.argmax(details["accuracy"])]`?

Reply:
Alright